### PR TITLE
Replace age field with DOB while Updating User

### LINKED ISSUE #7281: Inconsistency in details of user
**Is your feature request related to a problem? Please describe.**
When new user is registered, one of the info sought is the date of birth.
However, when an existing users edits/updates his details in his own profile, the edit profile form asks for Age and not date of birth.

There is inconsistency.

**Describe the solution you'd like**
On Edit Profile form (https://plexus.coronasafe.in/user/profile) ask for "Date of birth" instead of Age. 

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by rash-27**
## Proposed Changes

- Fixes #7281 

- This also requires the addition of a date of birth field in the user model (and removing age as it is not used anywhere)
- pr https://github.com/coronasafe/care/pull/1935 in the backend is for required backend changes

Screenshot 2024-03-02 043551

Screenshot 2024-03-02 043612

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
@rash-27 you need to modify the cypress test, as it's failing

**Comment by rash-27** [comment]
Screenshot from 2024-03-10 13-14-23
Screenshot from 2024-03-10 13-13-11
@nihal467 please review it (tests are modified based on change with the backend pr #1935)

File: cypress/e2e/users_spec/user_creation.cy.ts [hunk 1]
```diff
@@ -37,7 +37,7 @@ describe("User Creation", () => {
     "Please select the User Type",
     "Please enter valid phone number",
     "Please enter the username",
-    "Please enter date in YYYY/MM/DD format",
```
Comment (nihal467) [inline cypress/e2e/users_spec/user_creation.cy.ts:40]: revert this back

**Comment by nihal467** [comment]
image

@rithviknishad keep the same error message, don't change it, as it is more specific

**Comment by nihal467** [comment]
image

add a '*' to the DOB field

**Comment by nihal467** [comment]
image

keep the reflection in DD/MM/YYYY

**Comment by nihal467** [comment]
@rash-27 dont push the changes, @rithviknishad is pushing it

**Comment by rash-27** [comment]
Okay

**Comment by khavinshankar** [comment]
merging it, approved by @nihal467

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**